cpu/samd5x/cpu.c

```diff
@@ -144,11 +144,31 @@ uint32_t sam0_gclk_freq(uint8_t id)
     }
 }
 
+void cpu_pm_cb_enter(int deep)
+{
+    if (deep) {
+        /* we can only use the buck converter if fast clocks are off */
+        sam0_set_voltage_regulator(SAM0_VREG_BUCK);
+    }
+}
+
+void cpu_pm_cb_leave(int deep)
+{
+    if (deep) {
+        /* switch back to LDO */
+        sam0_set_voltage_regulator(SAM0_VREG_LDO);
+    }
+}
+
```

Comment:
Should we keep this in this PR ? I thought you only want to enable it for SAMLxx for now ?

Reply:
I noticed I can always enable the buck converter in standby and deep sleep, so I might as well do it together with enabling the buck converter for saml1x and saml2x.

I dropped the patch that fixes wake from standby when the DFLL is used, as this is unrelated as to which voltage regulator is used.
It requires the `cpu_pm_cb_leave()` callback though.